url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/Utils/StringUtil.java

```diff
@@ -0,0 +1,628 @@
+		// use our default encoding rather than system default
+		return fromReader(new InputStreamReader(in, Constants.DEFAULT_ENCODING));
+	}
+
+	/** Reads in the entire contents of a file into a string */
+	public static String fromFile(String path) throws IOException {
+		return fromReader(new FileReader(path));
+	}
+
+	/**
+	 * Main whether a string is null or the empty string
+	 * @param s the string
+	 * @return true if s is null or the empty string
+	 */
+	public static boolean isNullString(String s) {
+		return s == null || "".equals(s);
+	}
+
+	/**
+	 * Compare two strings for equality or both null.
+	 * @param s1 string 1
+	 * @param s2 string 2
+	 * @return true if strings are equal or both null
+	 */
+	public static boolean equalStrings(String s1, String s2) {
+		if (s1 == null) {
+			return s2 == null;
+		} else {
+			return s1.equals(s2);
+		}
+	}
+
+	private static long gensymCtr = 0;
+
+	/**
+	 * Generate a unique string.
+	 * @param base the initial substring
+	 * @return a string consisting of the supplied initial substring and a
+	 * unique counter value.
+	 */
+	public static String gensym(String base) {
+		return base + gensymCtr++;
+	}
+
+	/**
+	 * Trim a hostname, removing "www." from the front, if present, and the TLD
+	 * from the end. If this would result in an empty string, the entire name is
+	 * returned.
+	 * @param a hostname
+	 * @return the trimmed hostname
+	 */
+	public static String trimHostName(String hostname) {
+		if (hostname == null) {
+			return null;
+		}
+		int start = 0;
+		if (hostname.regionMatches(true, 0, "www.", 0, 4)) {
+			start = 4;
+		}
+		int end = hostname.lastIndexOf('.');
+		if (end <= start) {
+			// if trimming www left nothing but TLD, return whole name
+			return hostname;
+		}
+		return hostname.substring(start, end);
+	}
+
+	/**
+	 * Parse a string as a time interval. An interval is specified as an integer
+	 * with an optional suffix. No suffix means milliseconds, s, m, h, d, w
+	 * indicates seconds, minutes, hours, days and weeks respectively.
+	 * @param the interval
+	 * @return interval in milliseconds
+	 */
+	// tk - extend to accept combinations: xxHyyMzzS, etc.
+	public static long parseTimeInterval(String str) {
+		try {
+			int len = str.length();
+			char suffix = str.charAt(len - 1);
+			String numstr;
+			long mult = 1;
+			if (Character.isDigit(suffix)) {
+				numstr = str;
+			} else {
+				numstr = str.substring(0, len - 1);
+				switch (Character.toUpperCase(suffix)) {
+				case 'S':
+					mult = Constants.SECOND;
+					break;
+				case 'M':
+					mult = Constants.MINUTE;
+					break;
+				case 'H':
+					mult = Constants.HOUR;
+					break;
+				case 'D':
+					mult = Constants.DAY;
+					break;
+				case 'W':
+					mult = Constants.WEEK;
+					break;
+				default:
+					throw new NumberFormatException(
+							"Illegal time interval suffix");
+				}
+			}
+			return Long.parseLong(numstr) * mult;
+		} catch (IndexOutOfBoundsException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)